src/core/server/legacy/config/ensure_valid_configuration.test.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ensureValidConfiguration } from './ensure_valid_configuration';
+import { getUnusedConfigKeys } from '../../../../legacy/server/config/get_unused_config_keys';
+import { configServiceMock } from '../../config/config_service.mock';
+
+jest.mock('../../../../legacy/server/config/get_unused_config_keys');
+
+describe('ensureValidConfiguration', () => {
+  let configService: ReturnType<typeof configServiceMock.create>;
+
+  beforeEach(() => {
```

Comment:
Test file is succinct because the actual testing is done in `getUnusedConfigKeys` tests. `ensureValidConfiguration ` is mostly a passthough